nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -323,16 +322,16 @@ public void updateWorkflowInstanceAfterExecution(WorkflowInstance instance, Work
         updateWorkflowInstanceWithTransaction(instance, action, childWorkflows, workflows, changedStateVariables);
       }
     } else {
-      updateWorkflowInstance(instance);
+      updateWorkflowInstance(instance, action.executionStart);
     }
   }
 
-  public int updateWorkflowInstance(WorkflowInstance instance) {
+  public int updateWorkflowInstance(WorkflowInstance instance, DateTime started) {
```

Comment:
Do not pass the `started` argument in since the `instance.started` already should contain the correct value. And if it doesn't then the caller should update it to be correct before updating the instance. There is nothing special in `started` compared to the other timestamp fields.

Reply:
Done